kedro/framework/cli/starters.py

```diff
@@ -22,22 +24,53 @@
     KedroCliError,
     _clean_pycache,
     _filter_deprecation_warnings,
+    _safe_load_entry_point,
     command_with_verbosity,
+    get_entry_points,
 )
 
 KEDRO_PATH = Path(kedro.__file__).parent
 TEMPLATE_PATH = KEDRO_PATH / "templates" / "project"
-
-_STARTER_ALIASES = {
-    "astro-airflow-iris",
-    "standalone-datacatalog",
-    "pandas-iris",
-    "pyspark",
-    "pyspark-iris",
-    "spaceflights",
-}
 _STARTERS_REPO = "git+https://github.com/kedro-org/kedro-starters.git"
 
+
+@define(order=True)
+class KedroStarterSpec:  # pylint: disable=too-few-public-methods
+    """Specification of custom kedro starter template
+    Args:
+        alias: alias of the starter which shows up on `kedro starter list` and `kedro new`
+        template_path: the path to git repository
+        directory: by default it will look for `name`, optionally override with custom directory.
+        origin: preserved field used by kedro internally to determine where the starter
+        comes from, users do not need to provide this field.
+    """
+
+    alias: str
+    template_path: str
+    directory: Optional[str] = None
+    origin: Optional[str] = field(init=False)
+
+
+_OFFICIAL_STARTER_SPECS = [
+    KedroStarterSpec("astro-airflow-iris", _STARTERS_REPO, "astro-airflow-iris"),
```

Comment:
@MerelTheisenQB Antony convinced me that the `kedro-starters` is a special case. For other users, it will be likely just 1 template, so they don't have to have a single repository with multiple templates. By default, the top level directory will be the template, `directory` is an optional argument if they need to organise multiple templates in folders.

Reply:
That makes sense! Consider me convinced as well 🙂